contracts/pallet-ibc/src/ics20/mod.rs

```diff
@@ -561,11 +563,370 @@ pub fn full_ibc_denom(packet: &Packet, mut token: PrefixedCoin) -> String {
 use ibc::applications::transfer::error::Error as Ics20Error;
 
 pub trait HandleMemo<T: Config> {
-	fn execute_memo(packet_data: &PacketData, receiver: T::AccountId) -> Result<(), Ics20Error>;
+	fn execute_memo(packet: &Packet) -> Result<(), Ics20Error>;
 }
 
 impl<T: Config> HandleMemo<T> for () {
-	fn execute_memo(_packet_data: &PacketData, _receiver: T::AccountId) -> Result<(), Ics20Error> {
+	fn execute_memo(_packet: &Packet) -> Result<(), Ics20Error> {
+		Ok(())
+	}
+}
+
+pub trait SubstrateMultihopXcmHandler {
+	type AccountId;
+
+	fn transfer_xcm(
+		from: Self::AccountId,
+		to: Self::AccountId,
+		para_id: Option<u32>,
+		amount: u128,
+		currency: u128,
+	) -> Option<()>;
```

Comment:
pallet ibc does not know about Error type composable side.
So Result<(), ()> is equal Option<()>
I decided to have the Option<()> because in case if none centauri will emit event that error happened.